app/services/ubiquity/json_import/add_files.rb

```diff
@@ -116,36 +116,28 @@ def create_hyrax_uploaded_file(file_io, file_name)
 
     def avoid_duplicates_when_file_title_exist_in_work
       if @file.class == String
-        new_files_titles = check_if_file_titles_are_new
-        return  @hyrax_uploaded_file.clear if  @hyrax_uploaded_file.first.file.file.filename == new_files_titles
-        @hyrax_uploaded_file
+        ensure_single_file_is_not_duplicate
       elsif @file.class == Array
-        remove_nil_hash_from_array
+        remove_duplicate_hash_from_array
       end
     end
 
-    def remove_nil_hash_from_array
-      new_files_titles = check_if_file_titles_are_new
-      hyrax_uploaded_objects = @hyrax_uploaded_file.map { |hash| hash['path']}
-      if new_files_titles.present?
-        hyrax_uploads = hyrax_uploaded_objects.map {|carrierwave_object| carrierwave_object if new_files_titles.include?(carrierwave_object.file.file.filename)}
-        @hyrax_uploaded_file.reject! {|hash| not hyrax_uploads.include?(hash['path']) }
+    def ensure_single_file_is_not_duplicate
+      if  check_work_has_existing_file_title.present?
+        imported_files_titles = [@hyrax_uploaded_file.first.file.file.filename]
+        is_file_title_same = (imported_files_titles.first == check_work_has_existing_file_title.first)
```

Comment:
are we checking duplicate files based on their titles?

Reply:
yes